synapse/util/state_compressor.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2021 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from synapse.logging.context import defer_to_thread
+from synapse.metrics.background_process_metrics import run_as_background_process
+
+try:
+    import auto_compressor as state_compressor
+except ImportError:
+    state_compressor = None
+
+
+def setup_state_compressor(hs):
+    """Schedules the state compressor to run regularly"""
+
+    # Return if cannot import auto_compressor
+    if not state_compressor:
+        return
+
+    # Return if compressor isn't enabled
+    compressor_config = hs.config.statecompressor
+    if not compressor_config.compressor_enabled:
+        return
```

Comment:
We probably want to only run this only if `hs.config.workers.run_background_tasks` is true so that we don't run it on every worker?